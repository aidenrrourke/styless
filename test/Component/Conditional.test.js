import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test('Sets a complex variable in an if check', () => {
    const Div = styled.div`
		cursor: if(@disabled, not-allowed);
	`;
    const tree = renderer.create(<Div/>).toJSON()
    expect(tree).toMatchSnapshot();

    const tree2 = renderer.create(<Div disabled/>).toJSON()
    expect(tree2).toMatchSnapshot();
});

test('Sets conditional margins', () => {
    const Div = styled.div`
		margin: if(@val, 6px);
	`;
    const tree = renderer.create(<Div/>).toJSON()
    expect(tree).toMatchSnapshot();

    const tree2 = renderer.create(<Div val/>).toJSON()
    expect(tree2).toMatchSnapshot();
});

test('Sets conditional margins with variables', () => {
    const Div = styled.div`
		margin: if(@val, @t, @f);
	`;
    const tree = renderer.create(<Div t="20px" f="0px"/>).toJSON()
    expect(tree).toMatchSnapshot();

    const tree2 = renderer.create(<Div t="20px" f="0px" val/>).toJSON()
    expect(tree2).toMatchSnapshot();
});