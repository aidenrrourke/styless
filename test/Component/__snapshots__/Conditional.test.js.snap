// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sets a complex variable in an if check 1`] = `
<div
  className=""
/>
`;

exports[`Sets a complex variable in an if check 2`] = `
.c0 {
  cursor: not-allowed;
}

<div
  className="c0"
  disabled={true}
/>
`;

exports[`Sets conditional margins 1`] = `
<div
  className=""
/>
`;

exports[`Sets conditional margins 2`] = `
.c0 {
  margin: 6px;
}

<div
  className="c0"
/>
`;

exports[`Sets conditional margins with variables 1`] = `
.c0 {
  margin: 0px;
}

<div
  className="c0"
/>
`;

exports[`Sets conditional margins with variables 2`] = `
.c0 {
  margin: 20px;
}

<div
  className="c0"
/>
`;
